### ISSUE TITLE
OPTEE TA cannot read `cntpct_el0` register

### ISSUE THREAD

**Issue by Yuhala** [OP-TEE/optee_client#351]
Hello, I have an issue with reading the `cntpct_el0` register in my OPTEE TA. I have built an OPTEE TA which invokes `read_cntpct()` directly as mentioned by @jforissier [here](https://github.com/OP-TEE/optee_os/issues/3513#issuecomment-571518060) .However, I encounter the following error during my TA build:
```
Error: selected processor does not support requested special purpose register -- 'mrs r2, cntpct_el0' 
```
Interestingly, I succeed in reading this counter without any error in OPTEE OS as well as a regular Linux kernel module on the same machine. 

Does anyone know how I could solve this issue?

Thanks,
Peterson

**Comment by jforissier** [comment]
Hi @Yuhala,
Since the error occurs during compilation, I suspect the compiler flags are incorrect and select the wrong CPU. Could you show the compile command that causes this error?

**Comment by Yuhala** [comment]
Hi @jforissier, thanks for the tip. I've fixed it and all works now.

My `TA_DEV_KIT_DIR` macro was pointing to a `32` bit build of OPTEE OS : `optee_os/out/arm/export-ta_arm32` instead of the `64` bit version which is supported by my system. This was originally caused by me building OPTEE OS for `arm32` instead of `arm64`. 

In case this sounds confusing to you (because I noted it works in OPTEE OS), I rebuilt OPTEE OS separately (after already installing it on the target) and my rebuild used the wrong version, but that's fixed now.

Thanks again.
